File: sources/com/sina/weibo/sdk/share/BaseActivity.java
package com.sina.weibo.sdk.share;

import android.app.Activity;
import android.content.pm.ActivityInfo;
import android.content.res.TypedArray;
import android.os.Build;
import android.os.Bundle;
import java.lang.reflect.Field;
import java.lang.reflect.Method;

public class BaseActivity extends Activity {
    public final boolean a() {
        boolean z;
        Exception e2;
        try {
            TypedArray obtainStyledAttributes = obtainStyledAttributes((int[]) Class.forName("com.android.internal.R$styleable").getField("Window").get((Object) null));
            Method method = ActivityInfo.class.getMethod("isTranslucentOrFloating", new Class[]{TypedArray.class});
            method.setAccessible(true);
            z = ((Boolean) method.invoke((Object) null, new Object[]{obtainStyledAttributes})).booleanValue();
            try {
                method.setAccessible(false);
            } catch (Exception e3) {
                e2 = e3;
            }
        } catch (Exception e4) {
            Exception exc = e4;
            z = false;
            e2 = exc;
            e2.printStackTrace();
            return z;
        }
        return z;
    }

    public void onCreate(Bundle bundle) {
        if (Build.VERSION.SDK_INT == 26 && a()) {
            try {
                Field declaredField = Activity.class.getDeclaredField("mActivityInfo");
                declaredField.setAccessible(true);
                ((ActivityInfo) declaredField.get(this)).screenOrientation = -1;
                declaredField.setAccessible(false);
            } catch (Exception e2) {
                e2.printStackTrace();
            }
        }
        super.onCreate(bundle);
    }

    public final void setRequestedOrientation(int i2) {
        if (Build.VERSION.SDK_INT != 26 || !a()) {
            super.setRequestedOrientation(i2);
        }
    }
}

File: plugin_apk/src/main/java/com/example/plugin_apk/BaseActivity.kt
package com.example.plugin_apk

import android.annotation.SuppressLint
import android.app.Activity
import android.content.Context
import android.content.pm.ActivityInfo
import android.content.res.AssetManager
import android.content.res.Resources
import android.os.Bundle
import android.util.Log
import java.io.File

internal const val PLUGIN_NAME = "plugin_apk-debug.apk"

/**
 * Day：2025/4/29 14:39
 * @author zhangyelei
 */
open class BaseActivity: Activity() {
    var wrappedContext: PluginContextWrapper? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        fixActivityIcon()
    }

    @SuppressLint("DiscouragedPrivateApi")
    private fun fixActivityIcon() {
        try {
            // 获取 plugin 的 ic_launcher 真实 ID
            val iconResId = resources.getIdentifier("plugin_ic_launcher", "mipmap", "com.example.plugin_apk")
            if (iconResId == 0) return

            // 反射 mActivityInfo
            val field = Activity::class.java.getDeclaredField("mActivityInfo")
            field.isAccessible = true
            val activityInfo = field.get(this) as ActivityInfo
            // 设置 icon
            activityInfo.icon = iconResId
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun attachBaseContext(newBase: Context?) {
        val pluginAssetManager = createAssetManager(newBase!!)
        val pluginResources = Resources(
            pluginAssetManager,
            newBase.resources.displayMetrics,
            newBase.resources.configuration
        )
        wrappedContext = PluginContextWrapper(newBase, newBase.classLoader, pluginResources, pluginAssetManager)
        super.attachBaseContext(wrappedContext)
    }

    private fun createAssetManager(context: Context): AssetManager? {
        Log.e("zyl", "创建createAssetManager")

        return try {
            val apkPath = "/storage/emulated/0/Android/data/com.baidu.demoopengl/files/plugin_apk-debug.apk"
            Log.e("zyl", "plugin createAssetManager $apkPath")
            val assetManager = AssetManager::class.java.newInstance()
            val addAssetPathMethod = AssetManager::class.java.getMethod("addAssetPath", String::class.java)
            addAssetPathMethod.invoke(assetManager, apkPath)
            assetManager
        } catch (e: Exception) {
            e.printStackTrace()
            null
        }
    }
}